modules/ding_availability/js/ding_availability.js

```diff
@@ -92,6 +92,9 @@
     $.each(entity_ids, function(index, entity_id) {
       // Reserve button
       var reserver_btn = element.parents('.ting-object:first, .material-item:first').find('a[id$=' + entity_id + '].reserve-button');
+      if (!reserver_btn.length) {
+        reserver_btn = $('.ting-object:first').find('a[id$=' + entity_id + '].reserve-button');
```

Comment:
'element' variable is forming of "Availability label" which is present on ting_object's default and 
search_result view modes, for teaser we don't have it I was forced to find a reservation link from present ting_object and to form new "reserver_btn" variable.